### PR TITLE
Docs overhaul: GitHub Discussions, add logo, new theme

### LINKED ISSUE #4916: Sidebar scrolling in docs not working and general adjustment ideas
I was too lazy for a PR but I a quickfixed the sidebar in Alabaster. It did not stay "in place" like we had it before. More adjustments might come in proper PR.

**_~Well, the fix does not work in master but it does work on my local docs build!~_**

Update: Actually the sidebar scrolling did work, probably there was some caching issue with my local browser, so that is [x] DONE

Anyway, we should take care of docs build and customization. Some ideas:

- Explicitely state Alabaster as the theme to use.
- Set the pic we have for the beetbox organization as the logo in the docs. (Or do we have a logo or does someone want to design one?
- Default to "latest" version of the docs
- ...I might add some more ideas here...

Might be related: https://github.com/beetbox/beets/issues/4912

_Originally posted by @JOJ0 in https://github.com/beetbox/beets/issues/4644#issuecomment-1728881928_

### PR DESCRIPTION
**Pull request by JOJ0**
## Description

Fixes #4916

- Encourage users/devs to go to GitHub discussions instead of Discourse.
- Add the beets logo.
- ~Make sure the Alabaster theme stays used.~
- ~Some customization attempts to the Alabaster theme.~
- ~And some bad news.....read my post below.~
- New Sphinx theme "pydata" with some "beetrooty" customizations

## To Do

- [X] Documentation
- [X] ~Changelog.~
- [X] ~Tests.~

## Preview

Screenshot 2023-10-20 at 16 07 49

### CODE REVIEW

**Comment by JOJ0** [comment]
@sampsyo @RollingStar @Serene-Arc 

The bad news is:
- The Alabaster theme is not very mobile friendly. There is no "pull down / hamburger" menu implemented. It's almost impossible to navigate in the docs properly. (Yes I do very often read them on my mobile. It is a major concern to my personal use of it.)
- Also on a desktop browser it is lacking proper sidebar navigation. It seems we are not allowed to dive deeper into the tree than the first 2 levels. For example check out the Configuration chapter and compare the two themes:
  - https://beets.readthedocs.io/en/latest/reference/config.html#replace
  - https://beets.readthedocs.io/en/1.6.0/reference/config.html#replace

File: README_kr.rst [hunk 1]
```diff
@@ -104,5 +104,5 @@ Read More
 `Adrian Sampson`_ 와 많은 사람들의 지지를 받아 Beets를 만들었다.
 돕고 싶다면 `forum`_.를 방문하면 된다.
 
-.. _forum: https://discourse.beets.io
+.. _forum: https://https://github.com/beetbox/beets/discussions/
```
Comment (sampsyo) [inline README_kr.rst:107]: Looks like this got a duplicated `https://`.

**Review by sampsyo — COMMENTED** [review]
Awesome; thanks for changing all these links! I found one nit to fix.

It's too bad about the mobile usefulness of Alabaster. That doesn't seem great! I can see what you mean when browsing, for example, https://requests.readthedocs.io/ on mobile. It's fine-ish for reading the current page, but there is no navigation—the sidebar is just completely hidden on mobile.

Ah well. Maybe we should just go back to the RTD theme? Even if it's less beautiful, it is more full featured. I guess we could browse other Sphinx themes too, but it doesn't necessarily seem like the most effective use of time to worry _too_ much about picking the perfect one:
https://sphinx-themes.org

**Comment by Serene-Arc** [comment]
I vote for changing the theme if it's causing this much angst. The visual continuity is less important than the practical use, which is the point of documentation. We should just get a new theme and go with that. New theme right around the time of our only release for like two years wouldn't cause too much confusion either.

**Comment by JOJ0** [comment]
Did you want to write

vote against changing...

typo?

**Comment by JOJ0** [comment]
> Did you want to write
> 
> vote against changing...
> 
> typo? 

Or maybe I misunderstood.

We have Alabaster now, you want to change it back to readthedocs. Right?

**Comment by JOJ0** [comment]
@Serene-Arc forget my last two comments! I should read more carefully before replying 😅 I think we are on the same page here. Functionality should come first! That's why the Alabaster theme already has lost the race for me, even though I love it visually. So since yesterday evening I'm thinking about either going back to readthedocs (because we know it works!) or finding something new.

@sampsyo since you handed over "the mighty beetroot" to me, there was no going back! 🤣 Have a look below why we can't go back to the readthedocs theme haha!

**Comment by JOJ0** [comment]
So I had a wonderful morning with playing around with Sphinx theming (and no shit) I really enjoyed it :-)

**readthedocs** with some customization - top left background defaulting to blue was a no-go with the beetroot with white background.

**Comment by JOJ0** [comment]
This is the **book** theme. From scrolling thru all the themes yesterday evening this was my favorite and I thought it could be our best choice. I  had to do some customization though since initially the font-size overall was too big for my taste. I really couldn't stand that, so now let me proudly present....base-font set a nuance smaller. I lilke it.

Screenshot 2023-10-21 at 11 00 06

**Comment by JOJ0** [comment]
After my first failed attempts getting my former favorite - the book theme - right, I learned that book is a close descendet from the **pydata** theme. After failing with book (at first) i gave this one a try and suprisingly I have to admit: This is my absolute favorite now! Even though it has loads of features, it still has a very clean look! Also dig the smaller logo on top left. Best feature is (next to the secondary sidebar we know from the book theme already) the "top level navigation" - it is super handy to have our main chapters on first sight sitiign there and switching between them is made so easy without having to use a sidebar at all.

So yeah, that would be my favorite! What do you guys say? @Serene-Arc @sampsyo @Steffo99 @RollingStar 

(I have the setup of all those themes each in a single commit now so actually we could even give one a try and redecide later on - switching now is easy. To really test it for mobile purposes we have to actually merge it, I can't think of another easy way)

Screenshot 2023-10-21 at 10 37 58

**Comment by Steffo99** [comment]
Looks nice and intuitive!

**Comment by sampsyo** [comment]
Awesome!!! That PyData theme looks really good—it's less retro-looking than the official RTD theme, and the navigation features look useful. I say go for it!

**Comment by Serene-Arc** [comment]
I think it's great!

**Comment by JOJ0** [comment]
@Serene-Arc or @sampsyo a little help required :-)

I can't get the test-docs job to run successfully. I've added the them package to the docs requirements in setup.py, but that doesn't seem to be enough. A local docs run tells me this:

```
tox -e docs
.pkg: _optional_hooks> python /Users/jojo/.pyenv/versions/beets310/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_sdist> python /Users/jojo/.pyenv/versions/beets310/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: get_requires_for_build_wheel> python /Users/jojo/.pyenv/versions/beets310/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: prepare_metadata_for_build_wheel> python /Users/jojo/.pyenv/versions/beets310/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: build_sdist> python /Users/jojo/.pyenv/versions/beets310/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
docs: install_package> python -I -m pip install --force-reinstall --no-deps /Users/jojo/git/beets/.tox/.tmp/package/4/beets-1.6.1.tar.gz
docs: commands[0]> sphinx-build -W -q -b html docs /Users/jojo/git/beets/.tox/docs/tmp/html

Theme error:
no theme named 'pydata_sphinx_theme' found (missing theme.conf?)
docs: exit 2 (0.94 seconds) /Users/jojo/git/beets> sphinx-build -W -q -b html docs /Users/jojo/git/beets/.tox/docs/tmp/html pid=59710
.pkg: _exit> python /Users/jojo/.pyenv/versions/beets310/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
  docs: FAIL code 2 (11.44=setup[10.50]+cmd[0.94] seconds)
  evaluation failed :( (11.58 seconds)
```

(I think the "missing theme.conf" error is misleading - at least up to my current research....)

Thanks in advance for any hints!

**Comment by JOJ0** [comment]
Got it! Adding to tox.ini `deps=...` was missing. Sometimes reading docs of tools helps ;-)

**Comment by sampsyo** [comment]
Wow; awesome that this worked out so well!!

I did find this transparent-background PNG lying around. I hope this fits the bill!
logo-square

**Comment by JOJ0** [comment]
> Wow; awesome that this worked out so well!!
> 
> I did find this transparent-background PNG lying around. I hope this fits the bill! logo-square

Perfect! Thanks a lot! I quickly exchanged the log in `master` d80d45974b0d331ecb36ec8888c9c07d6cf0b233


Now both light and dark theme use it.

I reconsidered about the favicon: The regular "readthedocs" favicon is being used and I think a lot of people are very used to it, so I think that is actually a good thing.

So IMO, all sorted out here :-)

**Comment by RollingStar** [comment]
Thank you! Some examples with the new theme:

https://beets.readthedocs.io/en/latest/index.html

https://beets.readthedocs.io/en/latest/faq.html

https://beets.readthedocs.io/en/latest/team.html

**Pull request merged**